polars/polars-lazy/src/tests/aggregations.rs

```diff
@@ -553,7 +553,7 @@ fn test_take_in_groups() -> PolarsResult<()> {
         .lazy()
         .sort("fruits", Default::default())
         .select([col("B")
-            .take(lit(Series::new("", &[0u32])))
+            .take(lit(0u32))
```

Comment:
Note that 
```
    let df = fruits_cars();

    let out = df
        .lazy()
        .sort("fruits", Default::default())
        .select([col("B")
            .take(lit(0u32))
            .over([col("fruits")])
            .alias("taken")])
        .collect()?;
```
results now in 
```
┌───────┐
│ taken │
│ ---   │
│ i32   │
╞═══════╡
│ 3     │
│ 5     │
└───────┘
```
(previously:
```
┌───────┐
│ taken │
│ ---   │
│ i32   │
╞═══════╡
│ 3     │
│ 3     │
│ 5     │
│ 5     │
│ 5     │
└───────┘
```
which I would expect from a partition over query).

So I am fixing one issue by introducing another. Maybe somebody can point me in a better direction here.

Reply:
I thought about this again and changed my mind. The solution taken till now isn't intuitive and not consistent, this PR fixes this.

Have a look at:
```rust

    let df = fruits_cars();

    let out = df
        .lazy()
        .sort("fruits", Default::default())
        .select([col("B")
            .take(lit(Series::new("", &[0u32, 1u32])))
            .over([col("fruits")])
            .alias("taken")])
        .collect()?;
```
results in (main and this PR):
```
┌───────┐
│ taken │
│ ---   │
│ i32   │
╞═══════╡
│ 3     │
│ 2     │
│ 5     │
│ 4     │
└───────┘
```
which consistent to `.take(lit(Series::new("", &[0u32])))`

Reply:
I am not sure about this. A `take` in a `window` function should raise an error. This will cause OOB.

Reply:
This test was explicitly added by you in [this commit](https://github.com/pola-rs/polars/pull/2517/files#diff-c09def58626c52cf5623e7532a6fb52712745342b0dcfa76c53e2bcba6721e50R488). Should I throw an error if `take` is used with `window` or shall we create a separate issue for this?

Reply:
Hmm.. I don't know. I can take a look later. This part of the project I am quite reluctant to make changes, so I will take a look later.